src/robot/utils/connectioncache.py

```diff
@@ -137,13 +152,14 @@ def __len__(self):
     def __bool__(self):
         return self.current is not self._no_current
 
-    def resolve_alias_or_index(self, alias_or_index):
-        for resolver in self._resolve_alias, self._resolve_index:
+
+    def _resolve_connection(self, alias_or_index):
+        for resolver in self._resolve_alias, self._resolve_index, self._is_connection:
             try:
                 return resolver(alias_or_index)
             except ValueError:
                 pass
-        raise ValueError("Non-existing index or alias '%s'." % alias_or_index)
+        raise ValueError(f"Non-existing index,alias or connection '{alias_or_index}'.")
```

Comment:
Space needed after a comma (i.e. `index, alias`, not `index,alias`). I'd be actually fine considering connections as aliases which would mean we could preserve the old error message.

Reply:
reverted to old error msg